### PR TITLE
CK60i support

### PR DESCRIPTION
**Pull request by Gondolindrim**
Add CK60i PCB support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Gondolindrim** [comment]
Rebased master as per zvecr's recommendation

**Comment by Gondolindrim** [comment]
Due to the long commit list of rebase, I force reset to QMK master and cherry-picked the pertaining commits, then force-pushed with lease. This should make the commit history tidier and cleaner

File: keyboards/ck60i/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#define _______KC_NO
```
Comment (drashna) [inline keyboards/ck60i/keymaps/default/keymap.c:17]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ck60i/ck60i.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "ck60i.h"
+
+void encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (zvecr) [inline keyboards/ck60i/ck60i.c:20]:
```suggestion
__attribute__((weak)) void encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/ck60i/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+MCU = STM32F072
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
```
Comment (zvecr) [inline keyboards/ck60i/rules.mk:11]:
```suggestion
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/ck60i/rules.mk (hunk 3 above)
Comment (Gondolindrim) [reply keyboards/ck60i/rules.mk:11]: Committed in 141a4902f

File: keyboards/ck60i/keymaps/default/keymap.c (hunk 1 above)
Comment (Gondolindrim) [reply keyboards/ck60i/keymaps/default/keymap.c:17]: Committed in 01fe17e08

File: keyboards/ck60i/ck60i.c (hunk 2 above)
Comment (Gondolindrim) [reply keyboards/ck60i/ck60i.c:20]: Committed in da2257690

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**